resolwe_bio/processes/import_data/metadata.py

```diff
@@ -0,0 +1,136 @@
+    )
+
+    if data.domain.has_continuous_class:
+        target_description = f"Regression; numerical class {missing_in_class}"
+    elif data.domain.has_discrete_class:
+        target_description = (
+            "Classification; categorical class "
+            f"with {len(data.domain.class_var.values)} values {missing_in_class}"
+        )
+    elif data.domain.class_vars:
+        target_description = (
+            "Multi-target; "
+            f"{len(data.domain.class_vars)} target variables "
+            f"{missing_in_class}"
+        )
+    else:
+        target_description = "Not defined"
+
+    return target_description
+
+
+class ImportMetadata(Process):
+    """Upload clinical metadata table in Orange format.
+
+    Orange can read files in native tab-delimited format, or can load
+    data from any of the major standard spreadsheet file types, like CSV
+    and Excel. Native format starts with the names of attributes with
+    prefixes that define atribute type (continuous, discrete, time,
+    string) and role (class, meta, ignore, instance weights). Prefixes
+    are separated from the attribute name with a hash sign (“#”).
+
+    Legacy format with three header rows is also supported. The first
+    row lists feature (column) names. The second header row gives the
+    attribute type and the third header line contains role information
+    to identify dependent features (class), irrelevant features (ignore)
+    or meta features (meta).
+
+    For more information see Orange
+    [documentation](https://orange-visual-programming.readthedocs.io/loading-your-data/index.html#header-with-attribute-type-information).
+
+    An example of native tab-delimited format can be downloaded
+    [here](http://file.biolab.si/datasets/sample-head.xlsx).
+
+    """
+
+    slug = "upload-clinical-metadata"
+    name = "Clinical metadata table"
+    process_type = "data:metadata:clinical"
+    version = "1.0.0"
+    category = "Import"
+    scheduling_class = SchedulingClass.BATCH
+    requirements = {
+        "expression-engine": "jinja",
+        "executor": {"docker": {"image": "resolwebio/orange:1.0.0"}},
+        "resources": {"cores": 1, "memory": 8192},
+    }
+    data_name = '{{ src.file|default("?") }}'
+
+    class Input:
+        """Input fields to process ImportMetadata."""
+
+        src = FileField(label="Table with clinical metadata")
+
+    class Output:
+        """Output field of the process ImportMetadata."""
+
+        table = FileField(label="Uploaded table")
+        n_samples = IntegerField(label="Number of samples")
+        features = StringField(label="Number of features")
+        target = StringField(label="Target class description")
+        n_metas = IntegerField(label="Number of meta attributes")
+
+    def run(self, inputs, outputs):
+        """Run the analysis."""
+
+        table_path = inputs.src.import_file(imported_format="extracted")
+
+        extensions = [".csv", ".tab", ".tsv", ".xlsx", ".xls"]
```

Comment:
A `.txt` would also be a common file suffix and need to be supported.

Reply:
`.txt` is not supported in Orange so I left it out to avoid guessing the delimiter. It would be possible determine delimiter using `csv` module and then renaming the file, but this is not a really robust method.